File: src/test/java/mkl/testarea/itext5/annotate/AddMultipleFileAttachments.java
/**
 * 
 */
package mkl.testarea.itext5.annotate;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;

import org.junit.BeforeClass;
import org.junit.Test;

import com.itextpdf.text.Document;
import com.itextpdf.text.DocumentException;
import com.itextpdf.text.Paragraph;
import com.itextpdf.text.pdf.PdfFileSpecification;
import com.itextpdf.text.pdf.PdfWriter;

/**
 * @author mkl
 */
public class AddMultipleFileAttachments {
    final static File RESULT_FOLDER = new File("target/test-outputs", "annotate");

    @BeforeClass
    public static void setUpBeforeClass() throws Exception {
        RESULT_FOLDER.mkdirs();
    }

    /**
     * <a href="https://stackoverflow.com/questions/77988848/itext-failing-to-attach-files-when-adding-more-than-60-attachments">
     * itext failing to attach files when adding more than 60 attachments
     * </a>
     * <p>
     * This test attaches 73 file attachments to a PDF, works as expected, so
     * cannot reproduce the issue of the OP.
     * </p>
     */
    @Test
    public void testAdd73Files() throws DocumentException, IOException {
        Document document = new Document();
        PdfWriter writer = PdfWriter.getInstance(document, new FileOutputStream(new File(RESULT_FOLDER, "73Attachments.pdf")));
        document.open();

        document.add(new Paragraph("Test file with 73 attachments"));

        for (int i = 0; i < 73; i++) {
            PdfFileSpecification fs = PdfFileSpecification.fileEmbedded(writer,
                    String.format("folder/file_%d.txt", i),
                    String.format("File %d.txt", i),
                    String.format("Contents of file %d", i).getBytes());
            fs.addDescription("specificname", false);
            writer.addFileAttachment(fs);
         }

        document.close();
    }
}

File: src/test/csharp/annotate/AddMultipleFileAttachments.cs
using iTextSharp.text.pdf;
using iTextSharp.text;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace iText5.Net_Playground.Annotate
{
    internal class AddMultipleFileAttachments
    {
        /// <summary>
        /// itext failing to attach files when adding more than 60 attachments
        /// https://stackoverflow.com/questions/77988848/itext-failing-to-attach-files-when-adding-more-than-60-attachments
        ///
        /// This test attaches 73 file attachments to a PDF, works as expected, so cannot reproduce the issue of the OP.
        /// 
        /// (ported from the Java test with the same name)
        /// </summary>
        [Test]
        public void Add73Files()
        {
            var resultFile = @"..\..\..\target\test-outputs\annotate\73Attachments.pdf";
            Directory.CreateDirectory(@"..\..\..\target\test-outputs\annotate");

            using (Document document = new Document())
            {
                PdfWriter writer = PdfWriter.GetInstance(document, File.Create(resultFile));
                document.Open();

                document.Add(new Paragraph("Test file with 73 attachments"));

                for (int i = 0; i < 73; i++)
                {
                    PdfFileSpecification fs = PdfFileSpecification.FileEmbedded(writer,
                            String.Format("folder/file_{0}.txt", i),
                            String.Format("File {0}.txt", i),
                            Encoding.UTF8.GetBytes(String.Format("Contents of file {0}", i)));
                    fs.AddDescription("specificname", false);
                    writer.AddFileAttachment(fs);
                }

                document.Close();
            }
        }
    }
}
